ui/page/dcrdex/dex_onboarding_page.go

```diff
@@ -1285,29 +1358,42 @@ func (pg *DEXOnboarding) checkForPendingBondPayment(host string) {
 		return
 	}
 
-	pg.newTier = 1
-	pg.currentStep = onBoardingStepWaitForConfirmation
-	pg.bondConfirmationInfo = &bondConfirmationInfo{
-		bondCoinID:       bond.CoinID,
-		requiredBondConf: uint16(bondAsset.Confs),
-		currentBondConf:  int32(bond.Confs),
-	}
+	waitForBondTx := func() {
+		pg.newTier = 1
+		pg.currentStep = onBoardingStepWaitForConfirmation
+		pg.bondConfirmationInfo = &bondConfirmationInfo{
+			bondCoinID:       bond.CoinID,
+			requiredBondConf: uint16(bondAsset.Confs),
+			currentBondConf:  int32(bond.Confs),
+		}
+
+		// Set fields required by pg.stepWaitForBondConfirmation page. Also See:
+		// pg.bondAmountInfoDisplay.
+		bondAssetType := convertAssetIDToAssetType(bondAsset.ID)
+		pg.bondServer.bondAssets = map[libutils.AssetType]*core.BondAsset{
+			bondAssetType: bondAsset,
+		}
+		pg.bondServer.url = xcHost
+		pg.bondSourceAccountSelector = components.NewWalletAndAccountSelector(pg.Load, bondAssetType)
+		ok := pg.bondSourceAccountSelector.SetSelectedAsset(bondAssetType)
+		if !ok { // impossible but can happen if user deletes wallet shortly after posting bonds.
+			pg.notifyError(values.String(values.StrNoWalletLoaded))
+			return
+		}
 
-	// Set fields required by pg.stepWaitForBondConfirmation page. Also See:
-	// pg.bondAmountInfoDisplay.
-	bondAssetType := convertAssetIDToAssetType(bondAsset.ID)
-	pg.bondServer.bondAssets = map[libutils.AssetType]*core.BondAsset{
-		bondAssetType: bondAsset,
+		pg.waitForConfirmationAndListenForBlockNotifications()
 	}
-	pg.bondServer.url = xcHost
-	pg.bondSourceAccountSelector = components.NewWalletAndAccountSelector(pg.Load, bondAssetType)
-	ok := pg.bondSourceAccountSelector.SetSelectedAsset(bondAssetType)
-	if !ok { // impossible but can happen if user deletes wallet shortly after posting bonds.
-		pg.notifyError(values.String(values.StrNoWalletLoaded))
+
+	dexClient := pg.AssetsManager.DexClient()
+	if dexClient.IsLoggedIn() {
+		waitForBondTx()
 		return
 	}
 
-	pg.waitForConfirmationAndListenForBlockNotifications()
+	dexPasswordModal := dexLoginModal(pg.Load, dexClient, waitForBondTx)
+	dexPasswordModal.SetDescription(values.String(values.StrLoginDEXForPendingBonds))
+	dexPasswordModal.SetNegativeButtonCallback(waitForBondTx) // We'll display a form for them to login.
```

Comment:
Both positive and negative callbacks do the same thing?

Reply:
Yes. The `waitForBondTx` prepares the `onBoardingStepWaitForConfirmation` page and starts a listener for the pending bond. IF the user logs in, will fall through to display the `onBoardingStepWaitForConfirmation`.